packages/validation/src/sarif/suppressions/plugins/FingerprintSuppressor.ts

```diff
@@ -0,0 +1,31 @@
+import {Suppression, SuppressionKind, ValidationResult} from '../../../common/sarif.js';
+import {FingerprintSuppression, Suppressor} from '../types.js';
+
+export class FingerprintSuppressor implements Suppressor<FingerprintSuppression> {
+  kind: SuppressionKind = 'external';
+
+  suppressions: FingerprintSuppression[] = [];
+
+  preload(s: FingerprintSuppression[]): Promise<void> {
+    this.suppressions = s;
+    return Promise.resolve();
+  }
+
+  suppress(problem: ValidationResult): Suppression[] | Promise<Suppression[]> {
+    const suppression = this.suppressions.find(s => {
+      return s.fingerprint && s.kind === this.kind && s.fingerprint === problem.fingerprints?.['monokleHash/v1'];
+    });
+
+    suppression && problem.suppressions?.push(suppression);
```

Comment:
I think the suppression engine already handles this for you. I tried to keep the suppressors pure functions.

I also had to read this expression twice to understand what it does. A simple if-statement should improve readability:
```
if (suppression) {
  problem.suppressions.push(suppression);
}
```

Reply:
Nice catch, you're right, the engine already handles that, I kinda just "copied" the redux-suppressor and adapted it but didn't really check what the engine does.